lib/ffmpeg/transcoder.rb

```diff
@@ -226,5 +240,31 @@ def convert_prefix_options_to_string(transcoder_prefix_options)
       prefix_options = "#{prefix_options} " if prefix_options.length > 0
       return prefix_options
     end
+
+    def calculate_interim_max_dimensions
+      max_width = @movie.width
+      max_height = @movie.height
+      # Find best highest resolution
+      @movie.unescaped_paths.each do |path|
+        local_movie = Movie.new(path)
+
+        # If the local resolution is larger than the current highest
+          max_width = local_movie.width if local_movie.width > max_width
+          max_height = local_movie.height if local_movie.height > max_height
```

Comment:
Slightly more concise would be `max_height = [local_movie.height, max_height].max`